### ISSUE TITLE
Scorpion

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Vehicles#13]
Issues and notes on the Scorpion from in-game testing

UT2004 Draw  - 1.0
UT3    Draw  - 1.2
UT3Port Draw - 1.0 / 1.10 - 1.08 I think this becuase I parked the Hellbender and Scropion next to each other on both games and this looks right

  - [Priority]  
  - [Try 1.08 to 1.10 Drawscale]  
  - [Constantly has a driving sound while sitting unmoving inside it instead of the engine hum, this is strange my stock UT2004 SPMA without UT3Vehicles loaded in any way has the exact same sound and problem and so does the stock UT2004 Hellbender, was it always like that and I never noticed until UT3 added an engine hum?] 
  - [Secondary]  
  - [Projectile on UT3 homes in on Mantas, Raptors, Vipers, Scavenger, Furies, and Cicadas (and it seemed like the Nemesis and Nightshade as well), port only seems to home in on Mantas]  
  - [After ejecting the port explodes after a short distance unlike UT3 where it explodes only on contact, on UT3 it's possible to get back in after ejecting if it doesn't hit anything and evenutally stops moving (at least I think I did this on VCTF Suspense but I could be completely wrong)]  
  - [Has lights in the right spot but they are not team colored and don't cast light, light is cast by the stock UT2004 Scorpion]  
  - [On UT3 hoverboard causes little to no impact, on UT2004 it slightly pushes it and causes it to go 2 wheeled for a second, using the Heavy Vehicles mutator can probably solve this and work as an optional while leaving it as is for those who prefer knocking vehicles around with the Shock Rifle like normal UT2004]  
  - [Missing lights on the sides in that little black brown square area]  
  - [Missing light on the lower back section under the tail light]  
  - [On UT3 it looks like the blades might be team colored]  
  - [Extra]  
 - [I get fall damage on UT2004 but not UT3 on ejection]

**Comment by HellDragon-HK** [comment]
Scorpion impact results

Scorpion Impact with Manta
-------UT3 Results--------
--Front Hit   - Fair knockback and Manta is turned 90 degrees, get damage indicators, esaily pushed
--Side Hit    - Little knockback and Manta turned 90 degrees, can only be turned when trying to push
--Back Hit    - Fair knockback and Manta is turned nearly 180 degrees, get daamage indicators, easily pushed
--Diagnol Hit - Little knocback and Manta is sent sliding to the side of the Scorpion with some turning, can be pushed but not as easy
------UT2004 Results------ 
--Front Hit   - About the same as UT3 but no damage taken, easily pushed
--Side Hit    - About the same as UT3 but no damage taken, easily pushed
--Back Hit    - Fair knockback and Manta is sent sliding across the ground without turning, easily pushed
--Diagnol Hit - About the same as UT3 but no damage taken, easily pushed
Extra Note: UT2004 Manta took damage and Scorpion did not, in UT3 Scorpion took damage and Manta took little or none (can't tell since PS3 omits the numbered health)

Scorpion Impact with Scorpion
-------UT3 Results--------
Front Hit   - Little knockback, other Scorpion went vertical and damage indicators were noticed, cannot be pushed
--Side Hit    - Driving Scorpion went vertical and hit Scorpion flipped on it's side, damage indicators noticed, cannot be pushed
--Back Hit    - Other Scorpion lifted off it's back wheels and turned before coming down, cannot be pushed
--Diagnol Hit - Both Scorpions tried to flip with the hit Scorpion succeeding, damage indicators noticed, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback but no lift, can be slowly pushed
--Side Hit    - Other Scorpion tries to flip on it's side, slightest tap causes other Scorpion to flip, cannot be pushed
--Back Hit    - Other Scorpion is lifted off it's back wheels fairly high, can be slowly pushed
--Diagnol Hit - Other Scorpion tried to flip like a side hit, trying to push it gave the same flip result
Extra Note: UT2004 Scorpion did not take damage and also has no impact sound

Scorpion Impact with Hellbender
-------UT3 Results--------
--Front Hit   - Little knockback, damage indicators noticed, cannot be pushed
--Side Hit    - Little knockback and Hellbender jumps into the air, damage indicators noticed, cannot be pushed
--Back Hit    - Little knockback and causes Hellbender's back wheel to lift off the ground, damage indicators noticed, cannot be pushed
--Diagnol Hit - Little knockback and either the Hellbender flips or the Scorpion does and sometimes both, damage indicators noticed, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback and Hellbender is lifted off front wheels, can be pushed
--Side Hit    - Fair knockback and Hellbender easily flips on it's side, cannot be pushed but sometimes can be flipped
--Back Hit    - Fair knockback and Hellbender is lifted off back wheels, can be pushed
--Diagnol Hit - Little knockback and Hellbender is lifted up and turned, can be pushed turned and flipped
Extra Note: In UT2004 Hellbender had impact sound but Scorpion did not, in UT2004 neither vehicle was damaged

Scorpion Impact with Stealthbender
-------UT3 Results--------
--Front Hit   - Little knockback, damage indication, cannot be pushed
--Side Hit    - Little knockback and Stealthbender flips on it's side and Scorpion goes vertical, damage indicators noticed, cannot be pushed
--Back Hit    - Little knockback and Stealthbender is lifted off back wheels, damage indicatitors noticed, cannot be pushed
--Diagnol Hit - Little knockback and Scorpion flipped on it's side and can probably even get flipped upside down, damage indicators noticed, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback and Stealthbender goes completely vertical and turns 90 degrees before landing, trying to push it causes it to go vertical flip on it's side and then hop across the ground
--Side Hit    - Stealthbender's flipping and then endlessly hopping across the ground makes it too difficult to test further
--Back Hit    - 
--Diagnol Hit - 

Scorpion Impact with Raptor
-------UT3 Results--------
--Front Hit   - Fair knockback and Raptor turns slightly, damage indicators noticed, cannot be pushed
--Side Hit    - Fair knockback and Raptor turns slightly, damage indicators noticed, cannot be pushed
--Back Hit    - Fair knockback, damage indicatitors noticed, cannot be pushed
--Diagnol Hit - Fair knockback and causes Raptor to spin 90 degrees, damage indicators noticed, can be turned but not pushed
------UT2004 Results------ 
--Front Hit   - Medium knockback and Raptor spins 90 degrees, easily pushed like it's on ice
--Side Hit    - Medium knockback and Raptor spins 90 degrees, easily pushed like it's on ice
--Back Hit    - Medium knockback and Raptor spins 90 degrees, easily pushed like it's on ice
--Diagnol Hit - Medium knockback and Raptor spins nearly 180 degrees, easily pushed and turned like it's on ice
Extra Note: UT2004 Raptor took a lot of damage and Scorpion took none, in UT3 Scorpion appears to take damage but Raptor doesn't or is minor, UT2004 Scorpion missing impact sound

Scorpion Impact with Cicada
-------UT3 Results--------
--Front Hit   - No knockback but does get turned, damage indication, cannot be pushed
--Side Hit    - Little knockback, damage indication, cannot be pushed
--Back Hit    - Little knockback and gets turned, damage indication, cannot be pushed
--Diagnol Hit - Little knockback and gets turned, damage indication, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback, easily  pushed
--Side Hit    - Little knockback and gets turned, easily pushed
--Back Hit    - Little knockback and gets turned, easily pushed
--Diagnol Hit - No knockback but gets turned, cannot be pushed but can be turned
Extra Note: UT2004 Cicada took a lot of damage and Scorpion took none, in UT3 Scorpion appeared to take damage, UT2004 Scorpion missing impact sound

Scorpion Impact with Hellfire SPMA
-------UT3 Results--------
--Front Hit   - Little knocback and slightly lifted off the front wheels, damage indication, cannot be pushed
--Side Hit    - No knockback and slightly lifted off the ground, damage indication, cannot be pushed
--Back Hit    - Little knockback, cannot be pushed
--Diagnol Hit - Little knockback and slightly lifted off the ground, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knocback and slightly lifted off the front wheels, can be pushed
--Side Hit    - Little knockback and Hellfire SPMA is lifted off the ground and turns in the air before coming back down, can be pushed
--Back Hit    - Little knockback and Hellfire SPMA is highly lifted off the ground, can be pushed
--Diagnol Hit - Little knockback and Hellfire SPMA is slightly lifted off the ground, trying to push it causes SPMA to flip over
Extra Note: SPMA tread wheels messed up again after side hit, UT2004 Scorpion did not take damage, in UT3 Scorpion seems to take damage

Scorpion Impact with Paladin
-------UT3 Results--------
--Front Hit   - No knockback and Paladin is slightly lifted off it's front wheels, damage taken, cannot be pushed
--Side Hit    - No knockback and Paladin tries to flip, damage taken, cannot be pushed
--Back Hit    - No knockback or lift up, damage taken, cannot be pushed
--Diagnol Hit - No knockback and Scorpion is deflected and nearly flipped, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback and Paladin is lifted off front wheels, can be slowly pushed
--Side Hit    - Little knockback and Paladin flips on it's side, cannot be pushed and Scorpion goes under it
--Back Hit    - Little knockback and Paladin is lifted off back wheels, can be slowly pushed
--Diagnol Hit - Same as frontal hit, can be pushed but usually lifts up and Scorpion goes under it
Extra Note: UT2004 Paladin took a fair amount damage and Scorpion did not, in UT3 only Scorpion seems to take damage, Scorpion impact sound missing

Scorpion Impact with Goliath
-------UT3 Results--------
--Front Hit   - No knockback but kind of lifts up a bit, damage taken, can be slowly pushed
--Side Hit    - No knockback, damage taken, cannot be pushed
--Back Hit    - No knockback, damage taken, cannot be pushed
--Diagnol Hit - No knockback, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback, can be pushed
--Side Hit    - Little knockback, can be pushed
--Back Hit    - Little knockback, can be pushed
--Diagnol Hit - Little knockback and Goliath spins a little, can be pushed
Extra Note: In UT2004 neither vehicle took damage, in UT3 Scorpion has damage indicators

Scorpion Impact with Leviathan
-------UT3 Results--------
--Front Hit   - No knockback, damage taken, cannot be pushed
--Side Hit    - No knockback, damage taken, cannot be pushed
--Back Hit    - No knockback, damage taken, cannot be pushed
--Diagnol Hit - No knockback, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback, can be pushed
--Side Hit    - Little knockback and slightly lifts off the ground, cannot be pushed
--Back Hit    - Little knockback, can be pushed
--Diagnol Hit - Little knockback, cane be pushed
Extra Note: UT2004 Leviathan took slight damage and Scorpion did not, Leviathan probably did not in UT3

Scorpion Impact with Viper
-------UT3 Results--------
--Front Hit   - Little knockback, damage taken, can be turned but not pushed
--Side Hit    - Little knockback, damage taken, cannot be pushed
--Back Hit    - Little knockback and Viper lifts up and spins a little, damage taken, cannot be pushed
--Diagnol Hit - Little knockback and Ciper lifts into the air, damage taken, can be pushed and turned
------UT2004 Results------ 
--Front Hit   - Medium knockback and Viper is sent spinning across the ground like it's on ice, easily pushed like it's on ice
--Side Hit    - Little to no knockback but spins, easily pushed like it's on ice
--Back Hit    - Medium knockback and Viper is sent spinning across the ground like it's on ice, easily pushed like it's on ice
--Diagnol Hit - Little knockback and Viper spins and Scorpion then goes on top of it, cannot be pushed but spins like it's on ice
Extra Note: UT2004 Viper took damage

Scorpion Impact with Scavenger
-------UT3 Results--------
--Front Hit   - Not really knocked back but gets pushed to the side a little, easily pushed and turned and legs move with it
--Side Hit    - Fair knockback, easily pushed and turned and legs move with it
--Back Hit    - Not really knocked back but gets pushed to the side a little, can be pushed and turned and legs move with it
--Diagnol Hit - Not really knocked back but gets pushed to the side a little, can be pushed and turned and legs move with it
------UT2004 Results------ 
--Front Hit   - pending being added to mod
--Side Hit    - pending being added to mod
--Back Hit    - pending being added to mod
--Diagnol Hit - pending being added to mod

Scorpion Impact with Nightshade
-------UT3 Results--------
--Front Hit   - Little knockback and spin, damage taken, can be pushed
--Side Hit    - Fair knockback and little spin, damage taken, can be pushed
--Back Hit    - Little knockback and slight spin, damage taken, can be pushed
--Diagnol Hit - Little knockback and Scorpion nearly flips, can be pushed
------UT2004 Results------ 
--Front Hit   - Medium knockback, easily pushed like it's on ice
--Side Hit    - Fair knockback and spins almost 90 degrees, easily pushed like it's on ice
--Back Hit    - Fair knockback and slight spin, easily pushed like it's on ice
--Diagnol Hit - Medium knockback and 90 degree spin, easily pushed like it's on ice
Extra Note: In UT3 Scorpion and Nightshade look about the same size, in UT2004 Nightshade is larger which indicated either the Scorpion is smaller than it should be or Nightshade is larger than it should be, Nightshade is missing impact sound, Nightshade in UT2004 took damage

Scorpion Impact with Fury
-------UT3 Results--------
--Front Hit   - Little knockback and slight turn, damage taken, cannot be pushed
--Side Hit    - Little knockback, damage taken, cannot be pushed
--Back Hit    - Little knockback, damage taken, cannot be pushed
--Diagnol Hit - Little knockback and Scorpion is deflected, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Medium knockback, easily pushed like it's on ice
--Side Hit    - Fair knockback, easily pushed like it's on ice
--Back Hit    - Little knockback and 90 degree spin, easily pushed like it's on ice
--Diagnol Hit - Medium knockback, trying it push makes it spin like it's on ice
Extra Note: UT2004 Fury took a lot of damage

Scorpion Impact with Nemesis
-------UT3 Results--------
--Front Hit   - No knockback, damage taken, cannot be pushed
--Side Hit    - No knockback, damage taken, cannot be pushed
--Back Hit    - No knockback, damage taken, cannot be pushed
--Diagnol Hit - No knockback, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - pending being added to mod
--Side Hit    - pending being added to mod
--Back Hit    - pending being added to mod
--Diagnol Hit - pending being added to mod

Scorpion Impact with Darkwalker
-------UT3 Results--------
--Front Hit   - No knockback and Scorpion is deflected and nearly flips, can be turned but not pushed
--Side Hit    - No knockback and Scorpion is deflected and nearly flips and legs move, can be turned but not pushed
--Back Hit    - No knockback and Scorpion is deflected and flips on it's side and legs move, can be turned but not pushed
--Diagnol Hit - No knockback and Scorpion goes vertical and a lot of leg movement, can be turned but not pushed
------UT2004 Results------ 
--Front Hit   - pending being added to mod
--Side Hit    - pending being added to mod
--Back Hit    - pending being added to mod
--Diagnol Hit - pending being added to mod

**Comment by Unre-Alex** [comment]
@HellDragon-HK Oh man, so much bump testing...you should be a game tester. That must have taken some time huh?

I got rid of the projectile sound by changing AmbientSound=Sound'WeaponSounds.BaseGunTech.BTranslocatorHoverModule' to AmbientSound=None in EONSScorpionEnergyProjectileRed.uc

**Comment by HellDragon-HK** [comment]
Still got a lot to go, I just need the Scavenger now and I can post my Raptor results as for testing I feel like I'm making up for getting the job with the HaloUT team only for my PC to have died and made it look like I only wanted early access....not my fault but still feel guilty about it

That was a translocator sound?

**Comment by HellDragon-HK** [comment]
I'm definitely noticing sound differences between 32 and 64 bit now that I'm aware of them, in 32 bit the turbo doesn't sound right but it does in 64 bit, and in 64 bit we have a constant driving sound instead of the engine hum where in 32 bit it works pretty close to UT3 with the engine hum and movement sound

**Comment by GreatEmerald** [comment]
Hrm, that's not good... Also, the Linux version is based on the 64-bit one, so we need to support that. I don't think there is a list of differences yet, though so far I've found that the shaders need to be much simpler to work on the 64-bit version (cannot combine combiners), which is why for instance the vanilla UDamage looks different between the two. That's definitely something to test, also with the new vehicle textures.

**Comment by HellDragon-HK** [comment]
The U-Damage looks different? I didn't know that, I'm using 32 bit now so I guess I'll see it eventually and won't need to report it.

Ok what do you need exactly because this is making things a bit complicated now and I need a guide for what I'm doing if possible

**Comment by GreatEmerald** [comment]
The vanilla UDamage, not the UT3 one. I can't do anything about that, just saying that it's there without mods as well.

Basically you should test with the 64-bit version. From what I can tell, whatever works on 64-bit also works on 32-bit and on Linux. So as long as it works there it should be fine. The 64-bit version is much more limited/strict in what it can do, but we need to make sure that those limits are adhered to to maximise compatibility. So it would be nice to know which bugs are there only in the 64-bit version, but a bug that's only present in the 64-bit version is still a bug that needs to be fixed.

Oh, by the way, there's also the X-Fi version of UT2004 that changes the sound system. That's a pretty fun version to test as well, even though I don't think it's worth targetting it, since it's very niche.

**Comment by HellDragon-HK** [comment]
Oh sorry missed the vanilla word there.
I didn't have good experience with the X-Fi patch from what I remember, even though I had/have an X-Fi card the sounds seemed like they got really watered down and wasn't easy to fix the issue if I remember right.
I've been up to this point testing only with 64 bit so I think I've got most of the issues found, any idea why when activating the turbo in 32 bit doesn't sound right but it does in 64?

**Comment by GreatEmerald** [comment]
Yeap, I have an X-Fi card specifically for UT2004 too. The X-Fi patch is not terribly useful but still fun to play around with every now and then.

I haven't looked into the sound differences between the versions yet, so I have no idea why. That needs some debugging. Perhaps it's a matter of the sound format.

**Comment by HellDragon-HK** [comment]
How do you install the X-Fi patch and remove it easily because I had to reinstall my UT to fix it and if you want me to test that then yeah I will sure.

I'll say in 32 bit it sounds either sped up or cut off as well as different pitch maybe, it's not right that's for sure.  I'll give it a closer listen and try a more accurate description when I go back in game.

**Comment by GreatEmerald** [comment]
Install it in a completely different folder. But like I said I don't think I want to target it, so you don't need to do that.

**Comment by HellDragon-HK** [comment]
Scorpion also seems to have smaller wheels although not as bad as the Hellbender problem, in UT3 they come up to my character's waist and in UT2004 with the same character model they come to the thigh. 

The body of the Scorpion looks like it might be right and it's the wheels that are the issue including it's inability to drive over nodes since it's so low to the ground which in UT3 looks like the bottom of the body is at the thigh and in UT2004 is around knee height maybe.

I can't attempt to change this in the editor without changing the entire body just like the Hellbender.  Also I noticed in the editor the never used hatch animations, is there any chance of restoring those for when the Scorpion is sitting empty like we saw in pre-release screenshots of UT3?

**Comment by Unre-Alex** [comment]
@GreatEmerald Could you disable the daredevil announcement? I don't mind seeing it, but hearing it all the time is not very UT3. If it helps the sounds are DareDevil in announcermale2k4 and crowd_cheer in ONSVehicleSounds-S.

**Comment by HellDragon-HK** [comment]
ChaosUT has an Ant-Daredevil mutator you could try

**Comment by GreatEmerald** [comment]
@Unre-Alex Not without a configuration option somewhere. I consider that a feature of UT2004.

**Comment by HellDragon-HK** [comment]
@GreatEmerald 
Is there a possibility of restoring the unused hatch animations for the Scorpion like we saw for the pre-release Scorpion in screenshots?  (In case you missed the question, if not sorry to repeat)

**Comment by GreatEmerald** [comment]
I don't really know when they're supposed to play, but if I knew, then sure, that wouldn't be hard.

**Comment by HellDragon-HK** [comment]
In screenshots, we saw it was open when empty so logically when someone gets in it would close, I've been trying to watch if it also happens with self destruct but haven't been able to see well enough yet.

**Comment by HellDragon-HK** [comment]
Got the hatch animation working, it's open when Scorpion is empty and closes when you get in with the character's head visible as the hatch closes so it's not a ghost driving and it also works on eject.

If anybody wants to check it out here you go https://drive.google.com/open?id=1Qumj_RZFLCIK6YCcUqM8taGy_7CSnaa5

**Comment by Unre-Alex** [comment]
it works, nice!

**Comment by GreatEmerald** [comment]
Again, please send a pull request.

**Comment by HellDragon-HK** [comment]
I noticed that the health amount at which smoke starts is much less in UT2004, in UT3 it starts a lot sooner and with different and more locations plus sparks, I added what I've seen to the check list in detail as much as possible

**Comment by HellDragon-HK** [comment]
@Unre-Alex Can you try this and see what you think, I can't tell but I think it helps get the Scorpion closer to the agility it has in UT3, then again it might be my imagination and it does nothing
add in defaultproperties
	
TurnDamping=0
SteerSpeed=800.00 //200.00

**Comment by HellDragon-HK** [comment]
Sigh sorry for the quad post but I felt this needs to be separate

Here is the updated sound package which includes all Collide, DestructionWarning, EngineLoop, and the 2 EngingeThrusts https://drive.google.com/open?id=1vBt0J76ByX2sOVR09lTxcKPWdstLQEGb because I don't know how to do a pull request on an actual file

I changed the UT3Scorpion.uc to have these lines in defaultproperties
//IdleSound=sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_EngineLoop01'
ImpactDamageSounds(0) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide01';
ImpactDamageSounds(1) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide02';
ImpactDamageSounds(2) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide03';
ImpactDamageSounds(3) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide04';
ImpactDamageSounds(4) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide01';
ImpactDamageSounds(5) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide02';
ImpactDamageSounds(6) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Collide03';
ExplosionSounds(0) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Explode02';
ExplosionSounds(1) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Explode02';
ExplosionSounds(2) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Explode02';
ExplosionSounds(3) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Explode02';
ExplosionSounds(4) = Sound'UT3A_Vehicle_Scorpion.Sounds.A_Vehicle_Scorpion_Explode02';

EngineLoop seems too quiet for use and the stock Scorpion sounds almost the same so I changed it back to unused like you guys did

The DestructionWarning, and EngineThrust sounds are unused, hopefully someone can hook them up to the self destruct and booster as I don't know how

**Comment by HellDragon-HK** [comment]
Sorry for another one in a row

@GreatEmerald 
Anything to do with the EONSScorpion classes are missing in the git archive, Alex had to send me the projectile class for the projectile sound fix and then the wav files that were missing, are those supposed to not be included in the archive or is this an oversight?

**Comment by Unre-Alex** [comment]
@GreatEmerald HellDragon found a way to disable the daredevil announcement. Hopefully it can be added to the Scorpion and the Hellbender. bDoStuntInfo=False

**Comment by HellDragon-HK** [comment]
I didn't actually test it, so it does work then?

**Comment by Unre-Alex** [comment]
yes it works!

**Comment by HellDragon-HK** [comment]
I think it should be left on the Scorpion, seems appropriate when you end up rolling the Scorpion over multiple times from the boost like I did in my JB-Coralation video...but like you said UT3 doesn't have it so it swings both ways in sense and style

**Comment by GreatEmerald** [comment]
Yes, it should be listed.
UT3HudOverlay should actually be in UT3Common, and I think it already is.
Restore the headlights.
Stunt announcements stay on, it's a feature of UT2004.

**Comment by HellDragon-HK** [comment]
When trying to extract classes from the Mod Manta to see how to get trail effects on it something about UT3Hud was extracted and broke UT3Vehicles btw, on compile I kept getting an error on the Nightshade and UT3HUD

**Comment by HellDragon-HK** [comment]
@GreatEmerald Speaking of announcements I don't think I've noticed any Fender Benders with UT3 vehicles how about you @Unre-Alex ?

**Comment by Unre-Alex** [comment]
Great, another thing that GE will readd that I will have to remove for myself...

**Comment by HellDragon-HK** [comment]
You don't like Fender Benders really?  Anyway have you noticed any because I don't think I have?

**Comment by Unre-Alex** [comment]
I don't like the sounds with the UT3 vehicles, I don't mind seeing info. No, I haven't heard any.

**Comment by HellDragon-HK** [comment]
The code has the sound and the info so what you really need is an alternate announcer that has an empty announcement for those

**Comment by HellDragon-HK** [comment]
Anybody ever try using First Person View in the UT3 Scorpion, try it heh heh :D

**Comment by HellDragon-HK** [comment]
Trying to use self destruct on the Power Core causes this every single time

UT2004 Build UT2004_Build_[2005-11-23_16.22]

OS: Windows NT 6.2 (Build: 9200)
CPU: GenuineIntel PentiumPro-class processor @ 4009 MHz with 4095MB RAM
Video: NVIDIA GeForce GTX 980 Ti (7653)

General protection fault!

History: KPerContactCB <- CopyContacts <- KHandleCollisions <- KUpdateContacts <- KTickLevelKarma <- TickAllActors <- ULevel::Tick <- (NetMode=0) <- TickLevel <- UGameEngine::Tick <- Level Icybo2 <- UpdateWorld <- MainLoop <- FMallocWindows::Free <- FMallocWindows::Realloc <- 10910191 0 FArray <- FArray::Realloc <- 0*2 <- FMallocWindows::Free

**Comment by Unre-Alex** [comment]
lol, I can confirm that, and it happens with beta 4 too.

**Comment by HellDragon-HK** [comment]
Added link for UT3 Drawscale ukx edit

**Comment by HellDragon-HK** [comment]
The Scorpion's wheels are having a similar but much less troublesome issue as the Paladin, they were already going through the ground a little but using the UT3 draw takes it all the way to the bottom of the hubcap nearly, reducing it to 1.15 helps get it back to normal enough it might be acceptable, I don't know, what's making them go through the ground anyway, Wheel Radius?

**Comment by GreatEmerald** [comment]
On eject height: this is derived from `EjectMomentum`. Try changing that in defaultproperties (I'm not sure what's the default as that's an EONS thing, but you can find it by using the console `get EONSScorpion  EjectMomentum`).

The way self-destruct works is that it destructs one second after you get ejected (see https://github.com/GreatEmerald/UT3Vehicles/blob/master/Classes/UT3Scorpion.uc#L207), no more, no less. The distance the Scorpion covers is entirely based on its previous velocity, so it will go further if you're going down a cliff and less if you're going up.

Ehh, I don't think we can do anything about the fall speed of the driver, at least without making it relatively permanent. Unless this "feather-fall" effect can be given until it self-destructs, hmm...

Not sure about how that bug works, why would it not self-destruct after ejecting first... That needs some investigation I guess.

Sure, you can make it fire Raptor rockets, but the problem is that the ball naturally arcs as it's affected by gravity, a rocket does not. It might be worth porting only the homing part of the Raptor rocket or AVRiL to the ball, though.

Yes, I remember the crash on the cores. I wonder, would changing this line: https://github.com/GreatEmerald/UT3Vehicles/blob/master/Classes/UT3Scorpion.uc#L61 to read this:
```UnrealScript
if (Other != None && (ONSPRV(Other) != None || ONSPowerCore(Other) != None )))
```
solve the issue. Potentially also put this into the lines above that are currently commented out too.

**Comment by HellDragon-HK** [comment]
The velocity was full speed boost on a flat road (ONS-Downtown port of WAR-Downtown), it shouldn't explode directly in front of me with like it hit something shout it (map portal maybe, I'm not a mapper and don't know how those work but I've heard of them, could they cause it)?

On eject low grav needs to be applied with perhaps a 2-3 second timer before starting or something...just a thought

The homing part is exactly what I meant, that's what's needed with a radius check before it starts homing if possible

I'll be sure to try that when I can and let you know on adding that line

**Comment by HellDragon-HK** [comment]
@GreatEmerald Just a thought but is there an enable/disable function available that we can use to hide the hatch (bone I would guess) at 50% damage or so to make driver taking damage reasonable, I guess explosions would still be felt through it so it does make sense to take damage through the hatch, but it still would be a nice little detail if it could be done.

**Comment by HellDragon-HK** [comment]
@GreatEmerald "Yes, I remember the crash on the cores. I wonder, would changing this line: https://github.com/GreatEmerald/UT3Vehicles/blob/master/Classes/UT3Scorpion.uc#L61 to read this:

if (Other != None && (ONSPRV(Other) != None || ONSPowerCore(Other) != None))

solve the issue."

Nope same crash different map

I don't think there is anything different here in the log but just in case...
Critical: KPerContactCB
Critical: CopyContacts
Critical: KHandleCollisions
Critical: KUpdateContacts
Critical: KTickLevelKarma
Critical: TickAllActors
Critical: ULevel::Tick
Critical: (NetMode=0)
Critical: TickLevel
Critical: UGameEngine::Tick
Critical: Level Arctic Stronghold
Critical: UpdateWorld
Critical: MainLoop
Critical: FMallocWindows::Free
Critical: FMallocWindows::Realloc
Critical: 10910191 0 FArray
Critical: FArray::Realloc
Critical: 0*2
Critical: FMallocWindows::Free
Exit: Executing UObject::StaticShutdownAfterError
Exit: Executing UWindowsClient::ShutdownAfterError
Log: Waiting for file streaming thread to finish...
Exit: OpenAL Audio subsystem shut down.
Localization: No localization: Window.IDDIALOG_CrashBox.IDC_CrashBox (int)
Exit: Exiting.
Log: FileManager: Reading 0 GByte 477 MByte 309 KByte 926 Bytes from HD took 0.548997 seconds (0.481997 reading, 0.067000 seeking).
Log: FileManager: 0.463998 seconds spent with misc. duties
Uninitialized: Name subsystem shut down
Uninitialized: Log file closed, 12/26/17 19:04:59

**Comment by HellDragon-HK** [comment]
@GreatEmerald Crashed on an empty enemy turret as well
UT2004 Build UT2004_Build_[2005-11-23_16.22]

OS: Windows NT 6.2 (Build: 9200)
CPU: GenuineIntel PentiumPro-class processor @ 4009 MHz with 4095MB RAM
Video: NVIDIA GeForce GTX 980 Ti (7653)

General protection fault!

History: CopyContacts <- KHandleCollisions <- KUpdateContacts <- KTickLevelKarma <- TickAllActors <- ULevel::Tick <- (NetMode=0) <- TickLevel <- UGameEngine::Tick <- Level Arctic Stronghold <- UpdateWorld <- MainLoop <- FMallocWindows::Free <- FMallocWindows::Realloc <- 10910191 0 FArray <- FArray::Realloc <- 0*2 <- FMallocWindows::Free

**Comment by HellDragon-HK** [comment]
Updated the Scorpion ukx back to it's original Z translation and the wheels are still above ground so it was indeed all the suspensions fault....the Scorpion feels good to me and might be ready for an actual binary upload here.

@Unre-Alex @100GPing100 can you guys give it a test run and give your thoughts, note the class needs to be updated and to save you guys the trouble of going through and copying the pull request here's my class file https://drive.google.com/open?id=1IdsNFWOSTjYxDDyEmzvhXyUUfKxtTvdT be sure to backup the old first just in case...or I guess rely on restoring it from git if need be...I still prefer backup files

Oh and I can't stop the camera from clipping when looking backward and up for nothing sorry, the best solution to this is borrow from Aerial View to use it's collision capable camera somehow

**Comment by GreatEmerald** [comment]
Hrm, so it feels like it crashes before it gets the chance to call the impact function. That's rather unfortunate. The only solution to that might be to slow down the boost speed or so...

**Comment by HellDragon-HK** [comment]
Is there another way to handle the self destruct, maybe doing it similar to how a space fighter crashes into a wall?   I ask this because it's the way it's done that is causing the Scorpion to not have driving into walls impact damage right?

**Comment by HellDragon-HK** [comment]
I'm having a new issue I don't remember noticing, on the first time deploying the blades after getting in they go back to their idle position after fully extending, this can be repeated by getting out and then back in, and it only appears to be on the very first time.

**Comment by GreatEmerald** [comment]
The crash issue is not due to the self-destruct, that works fine. Rather, it seems that it is due to ramming the core at high enough speeds that the game doesn't really understand what's happening. At least that's what I'm thinking; one way to test would be to decrease the speed and see what happens.

**Comment by HellDragon-HK** [comment]
New crash with self-destruct, while attempting to take out a Hellfire there was a slight hill and my Scorpion when right above the cannon as the SPMA fired and the game crashed, I have no clue exactly what happened, the only thing I can think of is that the Scorpion was essentially destroyed twice at the same time.....this is probably a rare and hard to replicate crash however.

**Comment by HellDragon-HK** [comment]
New crash with Scorpion SD, happens when an ONS Energy Turret gets hit and Alex also confirmed it happens for him as well

**Comment by HellDragon-HK** [comment]
I think I finally got a solution to the power core crash, here is my test results and opinion...

SD Crash test fix 1 results - Damage = 100.0 & Momentum = 20 works but SD has no distance

SD Crash test fix 2 results - Damage = 100.0 & Momentum = 100 works, no distance and single hit took core from 100% to 18%

SD Crash test fix 3 results - Damage = 100.0 & Momentum = 120 works, core droped to 15%, no distance

SD Crash test fix 4 results - Damage = 100.0 & Momentum = 130 works, no distance, core dropped to 74%

SD Crash test fix 5 results - Damage = 150.0 & Momentum = 130 works, core dropped to 51% from 100%

SD Crash test fix 6 results - Damage = 200.0 & Momentum = 50 works, core dropped to 5% from 100%

Final opinion - distance isn't possible to keep and results 5 & 6 are too overpowered so result 4 seems best and is what I've put in the pull request.  You are able to kill a full 900 health tank in a single hit still so seems decent to me.

**Comment by HellDragon-HK** [comment]
Anyone who is play testing feel like the self destruct still might be a little too powerful compared to UT3, I think I took out a full or near full health Levi with a single Scorpion which I'm pretty sure isn't quite right but need to double check to be sure.
